tests/suites/os/tests/internet-sharing/index.js

```diff
@@ -0,0 +1,110 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+module.exports = {
+    title: 'Internet connection sharing tests',
+    tests: [
+        {
+            title: 'Internet sharing iptables rules test',
+            run: async function(test) {
+                // Test for the NetworkManager patch done in
+                // https://github.com/balena-os/meta-balena/pull/2963
+                //
+                // A racing condition between balenaEngine and NetworkManager led to
+                // some Internet connection sharing iptables rules not being applied
+                // when NM connection profile with Internet sharing was activated at boot.
+                //
+                // This test checks whether all necessary iptables rules are added by
+                // artificially blocking iptables for 1 second and then for 3 seconds
+                // while a connection profile with Internet sharing is activated.
+
+                // Counts all iptables rules set for Internet sharing by NM
+                async function countSharedIptables(that) {
+                    // Executes a command that returns an integer
+                    async function execInt(command, that) {
+                        const output = await that.worker.executeCommandInHostOS(
+                            command,
+                            that.link
+                        );
+                        return parseInt(output);
+                    }
+
+                    let count = await execInt(
+                        'iptables -S | grep -c nm-shared-dummy0 || true',
+                        that,
+                    );
+                    count += await execInt(
+                        'iptables -S -t nat | grep -c nm-shared-dummy0 || true',
+                        that,
+                    );
+                    count += await execInt(
+                        'iptables -S nm-sh-fw-dummy0 | wc -l',
+                        that,
+                    );
+                    count += await execInt(
+                        'iptables -S nm-sh-in-dummy0 | wc -l',
+                        that,
+                    );
+
+                    that.log(`iptables count ${count}`);
+
+                    return count;
+                }
+
+                // Create a dummy NM connection with Internet sharing enabled. 
+                await this.worker.executeCommandInHostOS(
+                    'nmcli c add type dummy ifname dummy0 con-name dummy' + 
+                    ' autoconnect no ipv4.method shared ipv6.method disabled',
```

Comment:
```suggestion
                    `nmcli c add type dummy ifname dummy0 con-name dummy 
                    autoconnect no ipv4.method shared ipv6.method disabled`,
```

Any reason why you chose to go interpolate these 2 strings except you wanting them to be pretty? You can instead use ` backticks to add multiline strings in JS? Disregard if there is some other reason.

Reply:
I did not know that multiline strings can be specified with backticks. Will change that now!